includes/payment-methods/class-wc-stripe-express-checkout-element.php

```diff
@@ -233,7 +233,7 @@ public function javascript_params() {
 			'is_pay_for_order'           => $this->express_checkout_helper->is_pay_for_order_page(),
 			'has_block'                  => has_block( 'woocommerce/cart' ) || has_block( 'woocommerce/checkout' ),
 			'login_confirmation'         => $this->express_checkout_helper->get_login_confirmation_settings(),
-			'is_product_page'            => $this->express_checkout_helper->is_product(),
+			'is_product_page'            => $this->get_is_product_page_for_ece(),
```

Comment:
Given that we're adding logic local to `WC_Stripe_Express_Checkout_Element`, it feels like we should add some tests to cover the new logic.